Let Resource rollback if import throws exception

jarekwg opened the pull request:
If a db exception (such as `Key (name, primary_contact)=(squaberblabeergabatersh, Jenny Susans) already exists.` is thrown during Resource.import_data (when we try to save the instance), then attempting to rollback the database may result in
```TransactionManagementError “You can't execute queries until the end of the 'atomic' block” ```, causing import_data to throw an unhelpful transaction exception. With this fix the function will exit cleanly and the db error will be returned in one of the row_results as intended.

No time to add a failing test at the moment, but hopefully this is sufficient information to go on..

bmihelac commented:
Thanks for contributing @jarekwg 

Is it possible to add a test case for this change?

jarekwg commented:
Hi, no worries.
I rebased the branch and added a failing test.
I've also made an .isort.cfg file that I can push up if you like. (Useful for automating import cleaning)

bmihelac commented:
Thanks!

bmihelac commented:
>  I've also made an .isort.cfg file that I can push up if you like. (Useful for automating import cleaning)

yes please, add another PR. I did not work with https://github.com/timothycrosley/isort yet, but it looks nice.
